<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html lang="en-US" xml:lang="en-US" xmlns="http://www.w3.org/1999/xhtml">

<head>
<link rel="stylesheet" type="text/css" href="http://gtpb.igc.gulbenkian.pt/bicourses/GTPB2015style.css">
</link>
<title>
GTPB: ARANGS15 Automated and reproducible analysis of NGS data - Home
</title>
</head>

<body>

<!-- GTPB header -->

<table width="910" align="center" >
<tr><td>
<img src="../images/GTPB2015logo.png" width="890" align="center" >
</td></tr>
</table>

	<table width="910" align="center" border="1">
		<tbody>
			<tr>
				<td>
<br>
					<font size="6" color="#006400"><b>ARANGS15</b></font>
					<h1>Automated and reproducible analysis of NGS data</h1>
					<b><a href="./../posters/ARANGS15.pdf">Downloadable poster in PDF</a><b><br>
				</td>
			</tr>
		</tbody>
	</table>

</td></tr>
</table>

<table width="910" align="center">
<font size="2">
<tr><td>
<p><b>&nbsp;&nbsp;&nbsp;IMPORTANT DATES</b> for this Course
<br />&nbsp;&nbsp;&nbsp;Deadline for applications: <b>April 22nd 2015</b>
<br />&nbsp;&nbsp;&nbsp;Latest notification of acceptance: <b>April 23rd 2015</b>
<br />&nbsp;&nbsp;&nbsp;Course date: <b> May 11th -  May 15th 2015</b>
</p>
<p><b>Candidates with adequate profile will be accepted in the next 72 hours after the application until we reach 20 participants.
</td></tr>
</font>
</table>



	<table width="910" align="center" border="1">
		<tbody>
			<tr>
				<td><h2>Instructors:</h2></td>
			</tr>

			<tr>
				<td>
					<img border="0" style="margin:4px" src="http://gtpb.igc.gulbenkian.pt/bicourses/images/rutger.jpg" width="91" align="left">
					<font size="2">
						<p align="justify">
							<b>Rutger Vos</b> studied biology at the University of Amsterdam, where he graduated in
							2000. He then embarked on his PhD research under professor Arne Mooers at Simon
							Fraser University in Vancouver, Canada, where he defended his thesis on
							phyloinformatic problems in 2006. As a self-taught programmer he then became
							involved in several open-source scientific software development projects while
							continuing his research career through a postdoctoral fellowship at the
							University of British Columbia (Vancouver, Canada) and a Marie Curie research
							fellowship at the University of Reading (Reading, UK). In Spring of 2012 he
							commenced his employment as the bioinformaticist of the Naturalis Biodiversity
							Center in a role where he combines novel research with
							bioinformatics contributions to various research programmes within the
							organization. In his spare time he also contributes to various open source
							software projects (<a href="http://treebase.org/">TreeBASE</a>, 
							<a href="http://search.cpan.org/dist/Bio-Phylo">Bio::Phylo</a>, 
							<a href="http://nexml.org/">NeXML</a>, <a href="http://www.supersmart-project.org">SUPERSMART</a>) and is co-PI of the
							PhyloTastic project. In addition Rutger has taught bioinformatics workshops in
							the US, Japan, China, Kenya and three times before at GTPB
							(PHYLOINF09,
							<a href="http://gtpb.igc.gulbenkian.pt/bicourses/2012/ARANGS12/">ARANGS12</a> and
							<a href="http://gtpb.igc.gulbenkian.pt/bicourses/2013/ARANGS13/">ARANGS13</a>).
							<br>
							<br>
							<b>Affiliation:</b> <a href="http://naturalis.nl/en/">Naturalis Biodiversity
							Center</a>, Leiden, the Netherlands
						</p>
					</font>
				</td>
			</tr>

			<tr>
				<td>
					<img border="0" style="margin:4px" src="http://gtpb.igc.gulbenkian.pt/bicourses/images/darin.jpg" width="91" align="left">
					<font size="2">
						<p align="justify">
							<b>Darin London</b> studied biology at Texas Tech University in the
							United States, graduating with a Masters of Science in 1999.  He has been
							supporting biological research with computational support for 12 years, starting
							at GlaxoSmithKline in 2000, then moving to the European Bioinformatics Institute
							to work on the Biomart system, and finally coming to the Institute for Genome
							Sciences and Policy at Duke University, Durham, USA in 2005.  He has developed
							automated analysis pipelines to help researchers analyze over 25Tb of Vertebrate
							NGS data (GAII and HiSeq), mostly involving research work on the Encyclopedia of
							DNA Elements (ENCode).  Darin has taught numerous workshops on programming and
							automation with the Perl programming language, including the phyloinformatics
							workshops at the NESCent, Durham, USA, and the GTPB
							(PHYLOINF09,
							<a href="http://gtpb.igc.gulbenkian.pt/bicourses/2012/ARANGS12/">ARANGS12</a> and
							<a href="http://gtpb.igc.gulbenkian.pt/bicourses/2013/ARANGS13/">ARANGS13</a>).<br>
							<br>
							<b>Affiliation:</b> <a href="http://www.genome.duke.edu/">Institute for Genome
							Sciences and Policy</a>, Duke University
							Medical Center, Durham (NC), USA.
						</p>
					</font>
				</td>
			</tr>
		</tbody>
	</table>

	<table width="910" align="center" border="1">
		<tbody>
			<tr>
				<td>
					<h2>Course Description</h2>
					<h3>Introduction</h3>
					Next generation sequencing (NGS) technologies for DNA have resulted in a yet
					bigger deluge of data. Researchers are learning that analyzing the data efficiently
					requires the creation of sophisticated pipelines, typically using commandline
					tools in a Linux or other Opensource Unix variant compute environment.  Many researchers
					have created these pipelines to successfully analyze their data.  Now they are faced
					with the challenge of making these pipelines available to their colleagues.  The issue
					of reproducibility has emerged as <a href="http://www.nature.com/nature/focus/reproducibility/">a major challenge</a>, as researchers, peer 
					reviewers, and companies in the life sciences discover that the software and data used 
					to produce a particular research finding are either not available, poorly documented, or 
					targeted to specific compute infrastructures that are not available to the wider research 
					community. To remedy this, funding agencies and journals are creating policies to promote 
					software reproducibility. In this brief workshop we will establish several <b>best 
					practices</b> of reproducibility in the analysis of data obtained by NGS. In doing so we 
					will encounter the commonly used <b>technologies</b> that enable these best practices by 
					working through <b>use cases</b> that illustrate the underlying principles. Building on the 
					basis of an existing pipeline of commandline utilities, we will illustrate how the entire 
					compute environment used to run the pipeline can be packaged into a unit that can be shared 
					with other researchers such that they can make full use of the environment on their own 
					machines, or on standard cloud compute environments such as amazon or rackspace.

					<h3>Best practices</h3>
					<ul>
						<!-- for example: bwa mapping and variant calling, de novo assembly, metagenomics -->
						<li>Commandline scripting of analysis steps</li>
						
						<!-- puppet/chef, docker -->
						<li>Provisioning systems to standardize software environment requirements</li>
						
						<!-- vagrant, disk images -->
						<li>Packaging of compute environment into static, portable units</li>
						
						<!-- vagrantboxes, docker hub -->
						<li>Sharing of compute environment packages</li>
					</ul>
					<h3>Technologies</h3>
					<ul>
						<li>Next generation sequencing platforms</li>
						<li>Command-line executables, command line scripting and batching</li>
						<li>Provisioning Systems: Puppet, Dockerfile</li>
						<li>Virtualization with Virtualbox and Vagrant</li>
						<li>Containerization with Docker</li>
					</ul>
					<h2>Target audience</h2>
					This course is aimed at researchers who've developed pipelines to analyze NGS
					data and now, faced with new reproducibility requirements, would like to learn
					how to package their analysis pipeline into in a reproducible (and shareable)
					way. This course will start with a very basic NGS pipeline that runs in a
					Linux commandline environment, and develop this pipeline into two packages that
					can be shared with, and used by other researchers. The ideal attendee is a scientist
					who is already comfortable developing scripted pipelines on the commandline, or who is
					not afraid to get his/her hands dirty to acquire the computer-literacy skills
					for dealing with the informatics side of data analysis.
					<h2>Pre-requisites</h2>
					The course assumes that attendees are not intimidated by the prospect of gaining
					experience working on UNIX-like operating systems (including the shell, and shell
					scripting).
					Attendees should understand some of the science behind high-throughput DNA sequencing
					and sequence analysis, as we will not go deeply into underlying theory (or the
					mechanics of given algorithms, for example) as such. What will be taught are
					technical solutions for automating and sharing such analyses in shareable, reusable compute
					environments, which will include (but is not limited to) beginner-level programming, and
					basic Linux provisioning. General computer literacy, (e.g. editing plain text data files, navigating
					the command line) will be assumed.
				</td>
			</tr>
		</tbody>
	</table>


<table width="910" align="center" >
<tr><td>
<br /><a href="./Application.html"><b>Application</b></a>
</td></tr>

<tr><td>
<br /><a href="./timetable.html"><b>Detailed Program </b></a>
</td></tr>
</table>


<table width="910" align="center">
<tr><td>

<p>
Instituto Gulbenkian de Ci&ecirc;ncia,</p>
<p>
Apartado 14, 2781-901 Oeiras, Portugal</p>
<p><a href="http://gtpb.igc.gulbenkian.pt/bicourses">GTPB Homepage</a></p>
<p><a href="http://www.igc.gulbenkian.pt">IGC Homepage</a></p>
<p>Last updated:&nbsp;&nbsp;<i><b>Mar 22nd 2015</b></i></p>
</td></tr>
</table>
</body>

</html>
